pkg/odo/cli/create/namespace/namespace.go

```diff
@@ -0,0 +1,139 @@
+package namespace
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"context"
+
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+
+	"github.com/redhat-developer/odo/pkg/log"
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions/clientset"
+	scontext "github.com/redhat-developer/odo/pkg/segment/context"
+)
+
+const RecommendedCommandName = "namespace"
+
+var (
+	createExample = ktemplates.Examples(`
+	# Create a new namespace
+	%[1]s my-namespace
+	`)
+
+	createLongDesc = ktemplates.LongDesc(`Create a new namespace.
+	This command directly performs actions on the cluster and doesn't require a push.
+	`)
+
+	createShortDesc = `Create a new namespace`
+)
+
+// NamespaceCreateOptions encapsulates the options for the odo namespace create command
+type NamespaceCreateOptions struct {
+	// Context
+	*genericclioptions.Context
+
+	// Clients
+	clientset *clientset.Clientset
+
+	// Parameters
+	namespaceName string
+
+	// Flags
+	waitFlag bool
+
+	// value can be either 'project' or 'namespace', depending on what command is called
+	commandName string
+}
+
+// NewNamespaceCreateOptions creates a NamespaceCreateOptions instance
+func NewNamespaceCreateOptions() *NamespaceCreateOptions {
+	return &NamespaceCreateOptions{}
+}
+
+func (o *NamespaceCreateOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
+}
+
+// Complete completes NamespaceCreateOptions after they've been created
+func (nco *NamespaceCreateOptions) Complete(cmdline cmdline.Cmdline, args []string) (err error) {
+	nco.namespaceName = args[0]
+	nco.Context, err = genericclioptions.New(genericclioptions.NewCreateParameters(cmdline))
+	if err != nil {
+		return err
+	}
+	if scontext.GetTelemetryStatus(cmdline.Context()) {
+		scontext.SetClusterType(cmdline.Context(), nco.KClient)
+	}
+	return nil
+}
+
+// Validate validates the parameters of the NamespaceCreateOptions
+func (nco *NamespaceCreateOptions) Validate() error {
+	return nil
+}
+
+// Run runs the namespace create command
+func (nco *NamespaceCreateOptions) Run(ctx context.Context) (err error) {
+	// Create the "spinner"
+	s := &log.Status{}
+
+	// If the --wait parameter has been passed, we add a spinner..
+	if nco.waitFlag {
+		s = log.Spinnerf("Waiting for %s to come up", nco.commandName)
+		defer s.End(false)
+	}
+
+	// Create the namespace & end the spinner (if there is any..)
+	err = nco.clientset.ProjectClient.Create(nco.namespaceName, nco.waitFlag)
+	if err != nil {
+		return err
+	}
+	s.End(true)
+
+	successMessage := fmt.Sprintf(`%s %q is ready for use`, strings.Title(nco.commandName), nco.namespaceName)
+	log.Successf(successMessage)
+
+	// Set the current namespace when created
+	err = nco.clientset.ProjectClient.SetCurrent(nco.namespaceName)
```

Comment:
Maybe the doc should mention this fact? Maybe it's just me, but I was actually surprised to see that my `kubeconfig` has been modified after running the command. I thought it would just create the namespace/project, just like `kubectl create namespace` :)

And if this is the expected behavior, it could be interesting to update the tests in `cmd_namespace_test` to check this too, I think.

Reply:
So as discussed together, this is the expected behavior. Resolving this then.